awd/helpers.py

```diff
@@ -174,67 +232,159 @@ def create_dss_message_body(
             }
         )
 
-    def delete(
-        self, sqs_base_url: str, queue_name: str, receipt_handle: str
-    ) -> EmptyResponseMetadataTypeDef:
-        """Delete message from SQS queue."""
-        logger.debug("Deleting %s from SQS queue: %s", receipt_handle, queue_name)
+    def delete(self, receipt_handle: str) -> EmptyResponseMetadataTypeDef:
+        """Delete message from SQS queue.
+
+        Args:
+            receipt_handle: The receipt handle of the message to be deleted.
+        """
+        logger.debug("Deleting %s from SQS queue: %s", receipt_handle, self.queue_name)
         response = self.client.delete_message(
-            QueueUrl=f"{sqs_base_url}{queue_name}",
+            QueueUrl=f"{self.base_url}{self.queue_name}",
             ReceiptHandle=receipt_handle,
         )
         logger.debug("Message deleted from SQS queue: %s", response)
         return response
 
-    def send(
+    def process_result_message(
         self,
-        sqs_base_url: str,
-        queue_name: str,
-        message_attributes: Mapping[str, MessageAttributeValueTypeDef],
-        message_body: str,
-    ) -> SendMessageResultTypeDef:
-        """Send message via SQS."""
-        logger.debug("Sending message to SQS queue: %s", queue_name)
-        response = self.client.send_message(
-            QueueUrl=f"{sqs_base_url}{queue_name}",
-            MessageAttributes=message_attributes,
-            MessageBody=str(message_body),
-        )
-        logger.debug("Response from SQS queue: %s", response)
-        return response
-
-    def receive(
-        self,
-        sqs_base_url: str,
-        queue_name: str,
-    ) -> Iterator[MessageTypeDef]:
-        """Receive message via SQS."""
-        logger.debug("Receiving messages from SQS queue: %s", queue_name)
+        sqs_message: MessageTypeDef,
+        retry_threshold: str,
+    ) -> None:
+        """Validate and then process an SQS result message based on content.
+
+        Args:
+            sqs_message: An SQS result message to be processed.
+            retry_threshold: The number of times to attempt processing an article.
+        """
+        if not self.valid_sqs_message(sqs_message):
+            raise InvalidSQSMessageError
+        doi = sqs_message["MessageAttributes"]["PackageID"]["StringValue"]
+        doi_process_attempt = DoiProcessAttempt.get(doi)
+
+        message_body = json.loads(str(sqs_message["Body"]))
+        receipt_handle = sqs_message["ReceiptHandle"]
+        self.delete(receipt_handle)
+        logger.exception("DOI: %s, Result: %s", doi, message_body)
+
+        if message_body["ResultType"] == "error":
+            doi_process_attempt.sqs_error_update_status(int(retry_threshold))
+        else:
+            doi_process_attempt.update_status(status_code=Status.SUCCESS.value)
```

Comment:
I dig it!

Looks like a special method for SQS errors, where you can pass the retry threshold.  Otherwise, use a more generic update status method with the SUCCESS value.